src/perform_step/low_storage_rk_perform_step.jl

```diff
@@ -182,3 +182,86 @@ end
 
   f(k, u, p, t+dt)
 end
+
+
+
+# 2R+ low storage methods
+function initialize!(integrator,cache::LowStorageRK2RPConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 1
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPConstantCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack A,Bl,Bhl,B,Bh,C = cache
+
+  k   = fsalfirst
+  integrator.opts.adaptive && (tmp = zero(uprev))
+
+  #stages 1 to s-1
+  for i in eachindex(A)
+    integrator.opts.adaptive && (tmp = tmp + (B[i] - Bh[i])*dt*k)
+    u = u + B[i]*dt*k
+    k = f(u + (A[i] - B[i])*dt*k, p, t + C[i]*dt)
+  end
+
+  #last stage
+  integrator.opts.adaptive && (tmp = tmp + (Bl - Bhl)*dt*k)
+  u   = u  + Bl*dt*k
+
+  #Error estimate
+  if integrator.opts.adaptive
+    atmp = calculate_residuals(tmp, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm, t)
+    integrator.EEst = integrator.opts.internalnorm(atmp,t)
+  end
+
+  integrator.k[1] = integrator.fsalfirst
+  integrator.fsallast = f(u, p, t+dt) # For interpolation, then FSAL'd
+  integrator.u = u
+end
+
+function initialize!(integrator,cache::LowStorageRK2RPCache)
+  integrator.kshortsize = 2
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.fsalfirst = cache.fsalfirst
+  integrator.fsallast = cache.k
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst, integrator.uprev, integrator.p, integrator.t)
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack k,tmp,atmp = cache
+  @unpack A,Bl,Bhl,B,Bh,C = cache.tab
+
+  @. k   = fsalfirst
+  integrator.opts.adaptive && (@. tmp = zero(uprev))
+
+  #stages 1 to s-1
+  for i in eachindex(A)
+    integrator.opts.adaptive && (@. tmp = tmp + (B[i] - Bh[i])*dt*k)
+    @. u = u + B[i]*dt*k
+    f(k, u + (A[i] - B[i])*dt*k, p, t + C[i]*dt)
```

Comment:
This line will create allocations during each call because of `u + (A[i] - B[i])*dt*k`. You should again pre-allocate some storage for that part.

Reply:
But won't the compiler do exactly the same thing?

Reply:
Making a separate allocation for this increases the memory allocations.
```
integ = init(prob_ode_large, alg,dt=1.e-2, save_start=false, save_end=false, save_everystep=false)
  Expression: Base.summarysize(integ) ÷ Base.summarysize(u0_large) <= 6
   Evaluated: 7 <= 6
ERROR: There was an error during testing

integ = init(prob_ode_large, alg, dt=1.e-2, save_start=false, save_end=false, save_everystep=false, alias_u0=true)
  Expression: Base.summarysize(integ) ÷ Base.summarysize(u0_large) <= 5
   Evaluated: 6 <= 5
ERROR: There was an error during testing
```
Earlier these worked for 6 and 5 respectively

Reply:
> But won't the compiler do exactly the same thing?

No. If you pre-allocate storage for that expression and store it in the cache, the allocation will only happen once during initialisation and the cache can be reused for every iteration. In its current state, the allocation will happen for every stage of every step. This will likely be slower.